src/bindings/node/node.cc

```diff
@@ -0,0 +1,236 @@
+    // reinterpret_cast<valhalla::tyr::actor_t*>(nativeObject)->~MyObject();
+    delete reinterpret_cast<valhalla::tyr::actor_t*>(nativeObject);
+  }
+
+ private:
+  explicit Actor(): actor(Configuration), env_(nullptr), wrapper_(nullptr) {
+  }
+
+  ~Actor() {
+    napi_delete_reference(env_, wrapper_);
+  }
+
+  static boost::property_tree::ptree Configuration;
+
+  static boost::property_tree::ptree json_to_pt(const std::string& json) {
+    std::stringstream ss;
+    ss << json;
+    boost::property_tree::ptree pt;
+    boost::property_tree::read_json(ss, pt);
+    return pt;
+  }
+
+  static boost::property_tree::ptree make_conf(const std::string config) {
+    return json_to_pt(config);
+  }
+
+  static napi_value New(napi_env env, napi_callback_info info) {
+    napi_status status;
+
+    napi_value target;
+    status = napi_get_new_target(env, info, &target);
+    if (status != napi_ok) {
+      napi_throw_error(env, NULL, "Failed to get new target");
+    }
+    bool is_constructor = target != nullptr;
+
+    if (is_constructor) {
+      // Invoked as constructor: `new MyObject(...)`
+      // currently not making use of any args, leaving in case we add some
+      size_t argc = 1;
+      napi_value args[1];
+      napi_value jsthis;
+      status = napi_get_cb_info(env, info, &argc, args, &jsthis, nullptr);
+      if (status != napi_ok) {
+        napi_throw_error(env, NULL, "Failed to parse constructor params");
+      }
+      // No need to get args to the constructor for now
+      Actor* obj = new Actor();
+
+      obj->env_ = env;
+      status = napi_wrap(env,
+                         jsthis,
+                         reinterpret_cast<void*>(obj),
+                         Actor::Destructor,
+                         nullptr,  // finalize_hint
+                         &obj->wrapper_);
+
+      if (status != napi_ok) {
+        napi_throw_error(env, NULL, "Failed to wrap Actor");
+      }
+      return jsthis;
+    }
+
+    // Invoked as plain function `MyObject(...)`, turn into construct call.
+    // NOTE: for now we could skip the args here because we dont accept any
+    // but we'll leave it in case we end up adding some later
+    size_t argc_ = 1;
+    napi_value args[1];
+    status = napi_get_cb_info(env, info, &argc_, args, nullptr, nullptr);
+    if (status != napi_ok) {
+      napi_throw_error(env, NULL, "Failed to parse constructor params");
+    }
+
+    const size_t argc = 1;
+    napi_value argv[argc] = {args[0]};
+
+    napi_value cons;
+    status = napi_get_reference_value(env, constructor, &cons);
+    if (status != napi_ok) {
+      napi_throw_error(env, NULL, "Failed to get reference value");
+    }
+
+    napi_value instance;
+    status = napi_new_instance(env, cons, argc, argv, &instance);
+    if (status != napi_ok) {
+      napi_throw_error(env, NULL, "Failed to create new instance");
+    }
+
+    return instance;
+  }
+
+  static napi_value Configure(napi_env env, napi_callback_info info) {
```

Comment:
all the configuration stuff should be at the module level, so not nested inside of the `Actor` class.